music21/musicxml/xmlToM21.py

```diff
@@ -1571,23 +1571,57 @@ def parse(self):
         # will be copied into the Score
 
         # copy spanners that are complete into the part, as this is the
-        # highest level container that needs them
-        rm = []
+        # highest level container that needs them. Ottavas are the exception,
+        # they should be put in the PartStaff that contains the first note
+        # in the Ottava.
+        completedSpanners = []
         for sp in self.spannerBundle.getByCompleteStatus(True):
-            self.stream.coreInsert(0, sp)
-            rm.append(sp)
+            if not isinstance(sp, spanner.Ottava):
+                # don't insert Ottavas, we'll do that after separateOutPartStaves().
+                self.stream.coreInsert(0, sp)
+            completedSpanners.append(sp)
         # remove from original spanner bundle
-        for sp in rm:
+        for sp in completedSpanners:
             self.spannerBundle.remove(sp)
         # s is the score; adding the part to the score
         self.stream.coreElementsChanged()
 
+        partStaffs: list[stream.PartStaff] = []
         if self.maxStaves > 1:
-            self.separateOutPartStaves()
+            partStaffs = self.separateOutPartStaves()
         else:
             self.stream.addGroupForElements(self.partId)  # set group for components (recurse?)
             self.stream.groups.append(self.partId)  # set group for stream itself
 
+        for sp in completedSpanners:
+            if not isinstance(sp, spanner.Ottava):
+                # non-Ottavas do not need to be inserted into a particular PartStaff,
+                # nor do they need to be filled.
+                continue
+            spannerPart: stream.Part | None = None
+            if partStaffs:
+                spannerPart = self.findFirstPartContaining(sp.getFirst(), partStaffs)
+            else:
+                spannerPart = self.stream
+
+            if spannerPart:
+                spannerPart.coreInsert(0, sp)
+                spannerPart.coreElementsChanged()
+                sp.fillIntermediateSpannedElements(spannerPart)
+
+    def findFirstPartContaining(
+        self,
+        obj: base.Music21Object,
+        parts: list[stream.Part]
+    ) -> stream.Part | None:
+        # This can be quite slow if the object is a long way into the first partstaff
+        # or in a second (or third?) partstaff.  I hope for a faster alternative.
+        for part in parts:
+            for mObj in part.recurse().getElementsByClass(obj.classes[0]):
+                if mObj is obj:
+                    return part
+        return None
```

Comment:
This should use `containerInHierarchy()` repeatedly until a Part is found.

Reply:
Done, but please review the code, since I ended up taking a short-cut that I _think_ will always work.